packages/block-library/src/navigation/index.php

```diff
@@ -194,7 +204,7 @@ function build_navigation_html( $block, $colors, $font_sizes, $level_zero = true
 		// End appending HTML attributes to anchor tag.
 
 		// Start anchor tag content.
-		$html .= '>';
+		$html .= '><span>';
```

Comment:
Is this `<span>` totally needed? Going to add a comment in the styles.

Reply:
We need to wrap the title and the icon separately [to apply alignment](https://github.com/WordPress/gutenberg/blob/add/nav-submenu-arrows/packages/block-library/src/navigation/style.scss#L44-L45). This span is for the title only. The HTML (simplified) output should look like this:

```html
<div class="item">
  <span>Products<span><span>˅</span>
</div>
```

Good idea to add the comment there 👍

Reply:
> We need to wrap the title and the icon separately to apply alignment. 

It doesn't seem to be related to `<span>`s elements but the anchor one.

> This span is for the title only.

Agree, but it doesn't seem to be necessary.

> The HTML (simplified) output should look like this:

well ... you are ignoring the fact that we are adding the CSS class to the span element which wraps the icon, and this is the key of my suggestion.

I've removed the span from the markup and changed the CSS selector using the class:

```scss
// Add some space between link text & submenu icon
& > a > .wp-block-navigation-link__submenu-icon {
 	margin-left: 4px;
}
```

<img width="844" alt="Screen Shot 2020-01-21 at 10 59 35 AM" src="https://user-images.githubusercontent.com/77539/72810864-3c444480-3c3d-11ea-8bd4-548022379fbc.png">

So for my point of view, adding the `<span>` element is not necessary.

Reply:
I agree that it's not necessary, but it's there because of the way I chose to have the item content (title & icon) aligned vertically. I wanted to be sure that the icon and title will remain vertically centered by default, so I used flexbox in `.wp-block-navigation-link__content`.